source/docs/flow_control.md

```diff
@@ -0,0 +1,182 @@
+### Overview
+
+Flow control in envoy is done by having limits on each buffer, and watermark callbacks.  When a
+buffer contains more data than the configured limit, the high watermark callback will fire, kicking
+off a chain of events which eventually informs the data source to stop sending data.  This back-off
+may be immediate (stop reading from a socket) or gradual (stop HTTP/2 window updates) so all
+buffer limits in Envoy are considered soft limits.  When the buffer eventually drains (generally to
+half of of the high watermark to avoid thrashing back and forth) the low watermark callback will
```

Comment:
The way I like to think of this is as a hysteresis loop. Don't need to change wording, but that's one possible way to explain it.